sync2jira/downstream_pr.py

```diff
@@ -0,0 +1,129 @@
+import sync2jira.downstream_issue as d_issue
+from sync2jira.intermediary import Issue, matcher
+
+
+log = logging.getLogger(__name__)
+
+
+def format_comment(pr, pr_suffix):
+    """
+    Formats comment to link PR.
+    :param sync2jira.intermediary.PR pr: Upstream issue we're pulling data from
+    :param String pr_suffix: Suffix to indicate what state we're transitioning too
+    :return: Formatted comment
+    :rtype: String
+    """
+    if 'closed' in pr_suffix:
+        comment = f"[merge request {pr.title}| {pr.url}] was closed."
+    elif 'reopened' in pr_suffix:
+        comment = f"[merge request {pr.title}| {pr.url}] was reopened."
+    elif 'merged' in pr_suffix:
+        comment = f"[merge request {pr.title}| {pr.url}] was merged!"
+    else:
+        comment = f"{pr.reporter} mentioned this issue in " \
+            f"[merge request {pr.title}| {pr.url}]."
+    return comment
+
+
+def comment_exists(client, existing, new_comment):
+    """
+    Checks if new_comment exists in existing
+
+    :param jira.client.JIRA client: JIRA Client
+    :param jira.resources.Issue existing: Existing JIRA issue that was found
+    :param String new_comment: Formatted comment we're looking for
+    :returns: Nothing
+    """
+    # Grab and loop over comments
+    comments = client.comments(existing)
+    for comment in comments:
+        if new_comment == comment.body:
+            # If the comment was
+            return True
+
+    return False
+
+
+def update_jira_issue(existing, pr, client):
+    """
+    Updates an existing JIRA issue (i.e. tags, assignee, comments etc).
+
+    :param jira.resources.Issue existing: Existing JIRA issue that was found
+    :param sync2jira.intermediary.PR pr: Upstream issue we're pulling data from
+    :param jira.client.JIRA client: JIRA Client
+    :returns: Nothing
+    """
+    # Format and add comment to indicate PR has been linked
+    new_comment = format_comment(pr, pr.suffix)
+    # Check if the comment if already there
+    if not comment_exists(client, existing, new_comment):
+        log.info(f"Added comment for PR {pr.title} on JIRA {pr.jira}")
+        client.add_comment(existing, new_comment)
+
+
+def sync_with_jira(pr, config):
+    """
+    Attempts to sync a upstream issue with JIRA (i.e. by finding
+    an existing issue or creating a new one).
+
+    :param sync2jira.intermediary.PR pr: PR object
+    :param Dict config: Config dict
+    :returns: Nothing
+    """
+    log.info("Considering upstream %s, %s", pr.url, pr.title)
+
+    # Create a client connection for this issue
+    client = d_issue.get_jira_client(pr, config)
+
+    # Check the status of the JIRA client
+    if not config['sync2jira']['develop'] and not d_issue.check_jira_status(client):
+        log.warning('The JIRA server looks like its down. Shutting down...')
+        raise JIRAError
+
+    # Find our JIRA issue if one exists
+    if isinstance(pr, Issue):
+        pr.jira = matcher(pr.content, pr.comments)
```

Comment:
Suggestion, name this `pr.jira_key ` to be clear about what it is.

Reply:
WFM